aiida/engine/launch.py

```diff
@@ -78,7 +78,14 @@ def run_get_pk(process: TYPE_RUN_PROCESS, *args: t.Any, **inputs: t.Any) -> Resu
     return runner.run_get_pk(process, *args, **inputs)
 
 
-def submit(process: TYPE_SUBMIT_PROCESS, wait: bool = False, wait_interval: int = 5, **inputs: t.Any) -> ProcessNode:
+def submit(
+    process: TYPE_SUBMIT_PROCESS,
+    inputs: dict[str, t.Any] | None = None,
+    *,
```

Comment:
It is not necessary, that is true, but I think it has a bunch of advantages:
* It is clearer to someone reading the submit call. With
  ```python
  submit(ArithmeticAddCalculation, inputs, True)
  ```
  You may ask, what is this `True`, whereas with
  ```python
  submit(ArithmeticAddCalculation, inputs, wait=True)
  ```
  it is a lot clearer.
* It gives us more flexibility for refactoring in the future. If we allow it to be positional, we cannot change this in the future without risking to break code. If we force it to be keyword only, we can change the signature more freely.

Reply:
Thanks @sphuber. Re (1), I agree that reading the `submit` call with the keyword argument is clearer, and enforcing this will make sure all `submit` calls have this readability. However, I'm not sure this weighs up against the fact the signature will be harder to understand for users that don't know about this `*` syntax (which will be most, I think):

```python
Signature:
engine.submit(
    process: 'TYPE_SUBMIT_PROCESS',
    inputs: 'dict[str, t.Any] | None' = None,
    *,
    wait: 'bool' = False,
    wait_interval: 'int' = 5,
    **kwargs: 't.Any',
) -> 'ProcessNode'
```

Re (2): perhaps, but if I understand correctly it would only give us "positional" flexibility, i.e. we could afterwards still put other positional arguments before `wait` if desired. Not sure if I see a lot of benefit there? 

Personally, for such a ubiquitously used function, I would prefer making it as simple and intuitive as possible.

Reply:
You don't _have_ to know about the syntax beforehand though. If you attempt to call it as positional, it will raise an exception:
```
TypeError: submit() takes 2 positional arguments but 3 were given
```
Besides, there are _lots_ of users that don't use docstrings to figure out how to use a method, but simply are following tutorials/documentation or are copying from existing code. So I think it won't be a problem.

But let's discuss this afternoon and see what others think.